x-pack/plugins/actions/server/create_unsecured_execute_function.ts

```diff
@@ -0,0 +1,160 @@
+}
+
+export interface ActionTaskParams extends Pick<ActionExecutorOptions, 'params'> {
+  actionId: string;
+  apiKey: string | null;
+  executionId: string;
+  consumer?: string;
+  relatedSavedObjects?: RelatedSavedObjects;
+}
+
+export type BulkUnsecuredExecutionEnqueuer<T> = (
+  internalSavedObjectsRepository: ISavedObjectsRepository,
+  actionsToExectute: ExecuteOptions[]
+) => Promise<T>;
+
+export function createBulkUnsecuredExecutionEnqueuerFunction({
+  taskManager,
+  connectorTypeRegistry,
+  isESOCanEncrypt,
+  preconfiguredConnectors,
+}: CreateBulkUnsecuredExecuteFunctionOptions): BulkUnsecuredExecutionEnqueuer<void> {
+  return async function execute(
+    internalSavedObjectsRepository: ISavedObjectsRepository,
+    actionsToExecute: ExecuteOptions[]
+  ) {
+    if (!isESOCanEncrypt) {
+      throw new Error(
+        `Unable to execute actions because the Encrypted Saved Objects plugin is missing encryption key. Please set xpack.encryptedSavedObjects.encryptionKey in the kibana.yml or use the bin/kibana-encryption-keys command.`
+      );
+    }
+
+    const connectorTypeIds: Record<string, string> = {};
+    const spaceIds: Record<string, string> = {};
+    const connectorIds = [...new Set(actionsToExecute.map((action) => action.id))];
+
+    const notPreconfiguredConnectors = connectorIds.filter(
+      (connectorId) =>
+        preconfiguredConnectors.find((connector) => connector.id === connectorId) == null
+    );
+
+    if (notPreconfiguredConnectors.length > 0) {
+      // log warning or throw error?
+    }
+
+    const connectors: PreconfiguredConnector[] = compact(
+      connectorIds.map((connectorId) =>
+        preconfiguredConnectors.find((pConnector) => pConnector.id === connectorId)
+      )
+    );
+
+    connectors.forEach((connector) => {
+      const { id, actionTypeId } = connector;
+      if (!connectorTypeRegistry.isActionExecutable(id, actionTypeId, { notifyUsage: true })) {
+        connectorTypeRegistry.ensureActionTypeEnabled(actionTypeId);
+      }
+
+      connectorTypeIds[id] = actionTypeId;
+    });
+
+    const actions = await Promise.all(
+      actionsToExecute.map(async (actionToExecute) => {
+        // Get saved object references from action ID and relatedSavedObjects
+        const { references, relatedSavedObjectWithRefs } = extractSavedObjectReferences(
+          actionToExecute.id,
+          true,
+          actionToExecute.relatedSavedObjects
+        );
+        const executionSourceReference = executionSourceAsSavedObjectReferences(
+          actionToExecute.source
+        );
+
+        const taskReferences = [];
+        if (executionSourceReference.references) {
+          taskReferences.push(...executionSourceReference.references);
+        }
+        if (references) {
+          taskReferences.push(...references);
+        }
+
+        spaceIds[actionToExecute.id] = actionToExecute.spaceId;
+
+        return {
+          type: ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE,
```

Comment:
Since this SO type is `namespaceType: 'multiple-isolated'`, do we need to also be specifying the `initialNamespaces` property when we create the saved object? Since we're using the bare repository, this won't be populated for you automatically.

https://github.com/elastic/kibana/blob/2a93a372ec8571e9c650bab6751af979e3d2274a/packages/core/saved-objects/core-saved-objects-api-server/src/apis/bulk_create.ts#L38-L48

Reply:
Looking at the `action_task_param` SO that's created by this, the namespaces field is set to `['default']` which seems ok? I can pass it in explicitly if it's necessary

Reply:
I don't know enough about the implications of these params existing in one space or another. By default, everything will live in the `default` space unless we explicitly say otherwise. For our immediate needs, this is might be ok? If a notification is triggered from a Case Assignment within the `marketing` space, would it be confusing that the corresponding notification was "sent" from the `default` space?

Reply:
@gsoldevila WDYT of this?

Reply:
@legrego it sounds like notifications are fairly space agnostic so there will not be an indication that the notification came from any particular space, even if the case assignment occurred in a specific space @gsoldevila is that a fair statement? Given that, I think we can default to the `default` space.

Reply:
I was going to ask exactly that. AFAIK the UI will not do anything at all with the action SO (which will only exist briefly), and the email notification will be space-agnostic. The notification message will include a backlink to the _Case_ in Kibana, but this backlink is generated by the _Cases_ logic, so I imagine the link itself will take the user to the correct space already. 

That being said, if we want to build a fully fledged notification center in the future, and we want to list pending / sent notifications, at that point it'll probably make sense to have the notifications linked to the right space.

@cnasikas any thoughts on this?

Reply:
This is correct. Cases will build the backlink to the case taking into account the space the case is created in. As for the task manager params SO, I think it is fine to be on the `default` space as it is not accessible by the user (is this assumption correct?) and it is controlled by the system.

Reply:
> I think it is fine to be on the default space as it is not accessible by the user (is this assumption correct?) and it is controlled by the system.

++ I agree. I wasn't sure if this would be surfaced in the Event Log or not, and whether or not we cared about that discrepancy within the Event Log.

Reply:
Good point. I am not sure either 🙂. Maybe @ymao1 can help with this.

Reply:
> The action executor will be writing event log entries for these executions. Do we need a way to differentiate between these actions and actions initiated by alerting?

For the event log, we already have a different `event.action` for when connectors are executed via http - `execute-via-http`, so I think having a separate one for this scenario would be good.  `execute-via-notification`?

Reply:
I suspect at some point the notification system will care about spaces, and we'll support non-pre-configured connectors that live across spaces, and we can defer figuring this out till then.

A non-pre-configured connector's space will show up in the event log, and presumably always will.  Pre-configured connectors likely appear to be in the "default" space today.  What happens when we have connectors in multiple spaces, not sure.  But that's all about the connector, and it sort of feels like the "space of the notification" is going to be a different thing.  If everything was supported I could use connector X that's defined in spaces A and B to perform a notification in space A, but the event log doesn't have a notion of "what space something is executed in".  It would probably need to be some new field, if we want to capture it.